go/vt/client/fakeserver_test.go

```diff
@@ -0,0 +1,233 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	mproto "github.com/youtube/vitess/go/mysql/proto"
+	"github.com/youtube/vitess/go/sqltypes"
+	"github.com/youtube/vitess/go/vt/topo"
+	"github.com/youtube/vitess/go/vt/vtgate/proto"
+	"github.com/youtube/vitess/go/vt/vtgate/vtgateservice"
+	"golang.org/x/net/context"
+)
+
+// fakeVTGateService has the server side of this fake
+type fakeVTGateService struct {
```

Comment:
The wrangler/testlib code is a good example of a library meant to be used by tests. I wasn't arguing against that, I just didn't want the fake to depend on bson RPC earlier. Ideally, fakes and test utility functions should be in their own libraries.

I would also caution against using too many layers in unit tests: if you're testing the client high level library, connecting through the low level library to the server, you can setup a fake server and connect to it, or you could use a fake client connector, and that would be simpler / easier, and make the unit test more 'unit'...